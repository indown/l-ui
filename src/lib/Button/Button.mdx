---
name: Button
---

import { Playground, Props } from 'docz'
import Button from './index'

# Button

<Props of={Button} />

## Basic usage

<Playground>
  <Button>No, click me</Button>
</Playground>