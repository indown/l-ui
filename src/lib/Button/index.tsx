import React from 'react';

interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
}

const Button:React.FC<ButtonProps> = (props) => {
    return(
        <button {...props}>{props.children}</button>
    )
}

export default Button;