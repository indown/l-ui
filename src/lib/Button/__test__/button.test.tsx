import Button from '../index'
import React from 'react';
import renderer from 'react-test-renderer';
import { mount, render } from 'enzyme'

describe('<Button />', () => {
  it('Button is a react component', () => {
    const component = renderer.create(
      <Button />
    );
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('Button be onClick', () => {
    const fn = jest.fn()
    const wrapper = mount(<Button onClick={fn}/>)
    wrapper.find('button').simulate('click')
    expect(fn).toBeCalledTimes(1)
  })
  it('Button has this className', () => {
    const wrapper = render(<Button className='test-classname'/>)
    expect(wrapper.hasClass('test-classname'))
  })
})