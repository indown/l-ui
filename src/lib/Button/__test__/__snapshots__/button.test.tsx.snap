// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Button is a react component 1`] = `<button />`;
